src/sql/src/plan/statement/ddl.rs

```diff
@@ -1407,7 +1411,217 @@ pub fn plan_create_views(
+                                        expr: Box::new(Expr::Identifier(vec![Ident::new(
+                                            "row_data",
+                                        )])),
+                                        positions: vec![SubscriptPosition {
+                                            start: Some(Expr::Value(Value::Number(
+                                                // LIST is one based
+                                                (i + 1).to_string(),
+                                            ))),
+                                            end: None,
+                                            explicit_slice: false,
+                                        }],
+                                    }),
+                                    data_type,
+                                },
+                                alias: Some(Ident::new(column.name.clone())),
+                            });
+                        }
+                        let query = Query {
+                            ctes: vec![],
+                            body: SetExpr::Select(Box::new(Select {
+                                distinct: None,
+                                projection,
+                                from: vec![TableWithJoins {
+                                    relation: TableFactor::Table {
+                                        name: RawName::Name(source_name.clone()),
+                                        alias: None,
+                                    },
+                                    joins: vec![],
+                                }],
+                                selection: Some(Expr::Op {
+                                    op: Op::bare("="),
+                                    expr1: Box::new(Expr::Identifier(vec![Ident::new("oid")])),
+                                    expr2: Some(Box::new(Expr::Value(Value::Number(
+                                        table_info.rel_id.to_string(),
+                                    )))),
+                                }),
+                                group_by: vec![],
+                                having: None,
+                                options: vec![],
+                            })),
+                            order_by: vec![],
+                            limit: None,
+                            offset: None,
+                        };
+
+                        viewdefs.push(ViewDefinition {
+                            name: view_name,
+                            columns: table_info
+                                .schema
+                                .iter()
+                                .map(|c| Ident::new(c.name.clone()))
+                                .collect(),
+                            with_options: vec![],
+                            query,
+                        });
+                    }
+                    let views = viewdefs
```

Comment:
Instead of first collecting all the viewdefs why don't we create a `let mut views = vec![];` vector and plan them one by one in the for loop above? Something like this:

```rust
  let mut viewdef = ViewDefinition { ViewDefinition<Raw>
      name: view_name,
      columns: table_info TableInfo
          .schema Vec<PgColumn>
          .iter() Iter<PgColumn>
          .map(|c| Ident::new(c.name.clone())) impl Iterator<Item = Ident>, &PgColumn
          .collect(),
      with_options: vec![],
      query,
  };

  views.push(plan_view(scx, &mut viewdef, &Params::empty(), temporary)?);
```

Reply:
My original plan had been to try and push these conversions out to the types but that ended up not working the way I hoped because the `SelectItem<Raw>` statements for the columns cannot be constructed from just a `PgColumn` struct but also require the offset of the column within the row. This ended up leaving the code in a somewhat intermediate state between the original primarily for loop based and my vision for a series of steps expressed as `iter().map().collect()` 

From a readability standpoint I think code like this is easier to understand when structured as a sequence of iterations over collections rather than big and potentially nested for loops, however as written this PR does not really achieve what I am going for. As I mentioned in Zoom I will massage this a bit more and either come up with something I think is better or make it more consistent in the style you suggest.